src/main/java/com/softserveinc/dokazovi/service/impl/PostServiceImpl.java

```diff
@@ -213,8 +214,10 @@ public Page<PostDTO> findAllByDirection(
     @Override
     public Page<PostDTO> findPostsByAuthorIdAndDirections(
```

Comment:
It's a good idea to rename the method, to make sure you don't accidentally skip some calls where id type got changed.

Reply:
By the way, while this is not applicable to your current design, some apps (especially based on domain-driven-development idioms, which you might see someday) declare dedicated classes for ID types, that allows type-checking the IDs you are passing to your methods.